libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1830,6 +1830,16 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 
 	if (exprType->category() == Type::Category::Contract)
 	{
+		// Warn about using address members on contracts
+		for (auto const& addressMember: IntegerType(160, IntegerType::Modifier::Address).nativeMembers(nullptr))
+			if (addressMember.name == memberName && *annotation.type == *addressMember.type)
```

Comment:
Funny enough, my second commit was "Simplify code" did the same (I think we had a chat to use `nativeMembers`), for some reason though that had 3 conditions:
```
+                       if (member.name == memberName && *member.type == *possibleMembers.front().type && member.declaration == nullptr)
```

I think the last `nullptr` check was just for safety?

Reply:
Yes, the type is a special type which should not be possible to be used directly.